### ISSUE TITLE
Incorrect unexpected-keyword-arg in dataclasses with inheritance

### ISSUE THREAD

**Issue by bdbernstein** [PyCQA/pylint#6275]
### Bug description

Looks like the generated init function that pylint creates doesn't work correctly (i.e., unexpected-keyword-arg) when there is a non-dataclass class it inherits from.
```
#hmm.py
from dataclasses import dataclass

class Wow:
    c: int
    def __init__(self, c: int):
        self.c = c

@dataclass
class Hmm(Wow):
    a: int = 9
    b: int = 7

    def __post_init__(self) -> None:
        super().__init__(self.b)

h = Hmm(a=1,b=2)
print(h.a,h.b,h.c)

```

### Command used

```shell
pylint hmm.py
```

### Pylint output

```shell
************* Module hmm
hmm.py:3:0: R0903: Too few public methods (0/2) (too-few-public-methods)
hmm.py:16:4: E1123: Unexpected keyword argument 'a' in constructor call (unexpected-keyword-arg)
hmm.py:16:4: E1123: Unexpected keyword argument 'b' in constructor call (unexpected-keyword-arg)
hmm.py:16:4: E1120: No value for argument 'c' in constructor call (no-value-for-parameter)
hmm.py:1:0: W0611: Unused cast imported from typing (unused-import)

--------------------------------------------------------------------
Your code has been rated at -3.08/10 (previous run: -3.08/10, +0.00)
```

### Expected behavior

No unexpected-keyword-arg

### Pylint version

```shell
pylint 2.9.3
astroid 2.6.2
Python 3.8.10 (default, Nov 26 2021, 20:14:08) [GCC 9.3.0]
```

### OS / Environment

Ubuntu 20.04.1 LTS

**Comment by jacobtylerwalls** [comment]
Duplicate of #3201, fixed in `2.10.0`.
